internal/server/api.yaml

```diff
@@ -386,7 +390,7 @@ components:
         type:
           $ref: '#/components/schemas/UploadTypes'
         options:
-          oneOf:
+          anyOf:
```

Comment:
What's the reason for this change? I believe that we indeed want `oneOf` here instead of `anyOf`.

Reply:
I agree that oneOf is preferable, the validator complains because AWSUploadRequestOptions and GCPUploadRequestOptions are identical except for the name.

Reply:
Interesting, how can I reproduce the validator's complaint? I would still prefer somehow suppressing the complaint to changing the schema because I believe that `anyOf` doesn't reflect well the reality.

Reply:
If you change it back to oneOf and run go test -v ./... you should get "{"errors":[{"detail":"request body has an error: doesn't match the schema: input matches more than one oneOf schemas","title":"400"}]}

Reply:
Hmm, we can change the spec like this: https://github.com/ondrejbudai/image-builder/commit/fb4451c2b6e6a4d502b0d1be45b40d27d472d756

Reply:
hmm, having some issues - did this work when you ran it?

Reply:
The problem here is that the aws and gcp upload requests share the exact same property names. So `AnyOf` might actually be correct from a purely syntactical pov. Logically there's a difference of course but well, the validator can't know that with just the spec in hand.

And the `type` enum is still a string and gets validated as such.

Reply:
Hm I was wrong about the enum type, the validator does consider it as more than a string. So moving oneOf up a level would work. I am a little hesitant for 2 reasons however:

1. I don't think it makes sense how the type turns into an enum with length of 1. Essentially acting as a constant, just to avoid the fact of having duplicate named fields. On top of that not using UploadTypes as well, the uploadtypes seems weird.
2. A more practical reason: oneOf/anyOf isn't really supported by the deepmap generator. So we have the type field set in the uploadrequest just to make parsing the options easier. Otherwise we'd need to parse the uploadrequest as a map first, then get the type, then parse it again as the correct upload request. 

So I'm not sure if it's worth it? Both options are not ideal I have to admit.

Reply:
I'm leaning towards using anyOf, oneOf is what we want but anyOf does apply accurately to our spec as we've defined it